File: IV/SDI (Systems for Design & Implementation)/Labs/Lab. 12/bookstore-core/src/main/java/ro/ubb/bookstore/core/model/Book.java
package ro.ubb.bookstore.core.model;

import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.Positive;
import java.util.Collections;
import java.util.HashSet;
import java.util.Objects;
import java.util.Set;
import java.util.stream.Collectors;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@ToString(callSuper = true, exclude = {"customerBooks"})
@Builder
public class Book extends BaseEntity<Long> {
    @Column(name = "isbn", nullable = false)
    @Getter
    @Setter
    private String isbn;

    @Column(name = "title", nullable = false)
    @Getter
    @Setter
    private String title;

    @Column(name = "author", nullable = false)
    @Getter
    @Setter
    private String author;

    @Column(name = "price", nullable = false)
    @Getter
    @Setter
    @Positive(message = "BookStore Exception -> Book's price must be positive.")
    private Float price;

    @OneToMany(mappedBy = "book", cascade = CascadeType.ALL, fetch = FetchType.EAGER, orphanRemoval = true)
    @Setter
    private Set<CustomerBook> customerBooks = new HashSet<>();

    public Set<CustomerBook> getCustomerBooks() {
        return customerBooks == null ? new HashSet<>() : customerBooks;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Book book = (Book) o;
        return isbn.equals(book.isbn);
    }

    @Override
    public int hashCode() {
        return Objects.hash(isbn);
    }
}

File: 4th semester/WP (Web Programming)/Labs/Lab8 ASP.NET/lab8_AspNet/lab8_AspNet/Models/Book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace lab8_AspNet.Models
{
    public class Book
    {
        public int id { get; set; }
        public string title { get; set; }
        public string author { get; set; }
        public string cathegory { get; set; }
        public float price { get; set; }

        public Book()
        {
            this.id = -1;
        }

        public Book(int id, string title, string author, string cathegory, float price)
        {
            this.id = id;
            this.title = title;
            this.author = author;
            this.cathegory = cathegory;
            this.price = price;
        }
    }
}